lib/Tomcat/ModjkTest.pm

```diff
@@ -0,0 +1,111 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Provide functionality for apache2 apache2-mod_jk installation
+# and configuration to test the functionality of interaction of apche2
+# and tomcat via apach2-mod_jk
+# accessing the tomcat manager page, and browsing tomcat examples using keyboard.
+# Maintainer: QE Core <qe-core@suse.de>
+
+package Tomcat::ModjkTest;
+use base "x11test";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+use registration;
+
+# allow a 60 second timeout for asserting needles
+use constant TIMEOUT => 60;
+
+sub mod_jk_setup() {
+    my $self = shift;
+
+    # install and configure tomcat in console
+    $self->select_serial_terminal();
+
+    record_info('install and configure apache2 and apache2-mod_jk connector Setup');
+    zypper_call('in apache2 apache2-mod_jk');
+
+    $self->conn_apache2_tomcat();
+    $self->load_jk_module();
+}
+
+sub func_conn_apache2_tomcat() {
+    my $self = shift;
+    # Connection from apache2 to tomcat: Functionality test
+    $self->select_serial_terminal();
+    systemctl('stop apache2');
+    systemctl('stop tomcat');
+    assert_script_run(
+        "echo  \"\$(cat <<EOF
+JkMount /*       ajp13
+JkMount /*.jsp   ajp13
+JkMount /*.xhtml ajp13
+EOF
+        )\"  >> /etc/apache2/conf.d/jk.conf", timeout => 180
+    );
+    systemctl('start apache2');
+    systemctl('status apache2');
+
+    systemctl('start tomcat');
+    systemctl('status tomcat');
+}
+
+sub conn_apache2_tomcat() {
+    # Connection from apache2 to tomcat: tomcat part
+    systemctl('stop tomcat');
+
+    # Define an AJP 1.3 Connector on IPv6 localhost port 8009
+    my $ajp_connector = qq(    <Connector protocol="AJP/1.3" address="::1" port="8009" redirectPort="8443" \/>);
+    assert_script_run qq(sed -i '/<\!-- Define an AJP 1.3 Connector on port 8009 -->/ a $ajp_connector\' /etc/tomcat/server.xml);
+
+    # Define a worker "ajp13" which listens behind the above AJP 1.3 Connector address and port
+    assert_script_run('cp /usr/share/doc/packages/apache2-mod_jk/workers.properties /etc/tomcat/workers.properties');
+    assert_script_run(
+        "echo \"\$(cat <<EOF
+worker.list=ajp13
+worker.ajp13.reference=worker.template
+worker.ajp13.host=localhost
+worker.ajp13.port=8009
+EOF
+        )\" >> /etc/tomcat/workers.properties", timeout => 180
+    );
+
+    # Connection from apache2 to tomcat: apache2 part
+    systemctl('stop apache2');
+    assert_script_run('cp -ai /usr/share/doc/packages/apache2-mod_jk/jk.conf /etc/apache2/conf.d');
+    if (is_sle('=15') or is_sle('<12-sp4')) {
+        record_soft_failure 'boo#1167896';
```

Comment:
This bug seems to be resolved, so this soft-failure makes no sense here. Also, there is no commenting here for the reason that this soft-failure exists and the description of the issue.

Reply:
I re-opened the bug, Issue is observed only in SLE 15 and SLE12 SP3/SP2.

Reply:
As I wrote in my comment above, it would be nice also to comment and explain the reason in the code of the soft-fail :)

Reply:
Sure, I added comment there too :) Thanks Anna